### DISCUSSION TITLE
Opt-out of 'Unsafe builtin method' warnings

### DISCUSSION THREAD

**Discussion by turbek** [gatsbyjs/gatsby#30760] [Help]
Hi,

After migrating to the 3.0 version we noticed that warnings appeared in our log while building the project.

> warn Unsafe builtin method was used, future builds will need to rebuild all pages
> warn WebpackError: Unsafe builtin usage fs.lstatSync

These warnings appear because we are using a local plugin for auditing so we are parsing the built HTML files and we'd like to dump our results in a directory. (so we need write and remove)

These warnings are there because Gatsby cannot ensure that the generated code didn't change between builds when 'fs' is used so it has to disable the 'Incremental Builds" feature. That's why it throws warnings.

My question is, since AFAIK the "Incremental Builds" is a Gatsby Cloud feature and our application is not using Gatsby Cloud. We'd like to disable these warnings. Is there a workaround for this issue or is it possible to turn these warnings off somehow?

Thanks,
Adam

**Comment by turbek** [comment]
I just found a PR discussing opt-out: https://github.com/gatsbyjs/gatsby/pull/30216

**Comment by LekoArts** [comment]
Incremental Builds is available for everyone, what Gatsby Cloud has are "Cloud Builds", so Incremetal Builds with Superpowers (even quicker). Thus the warning is relevant for everyone! :)

I think you misunderstood the comments in the PR, there probably won't be an option to opt out of these warnings (as otherwise you wouldn't know why it's not incremental) but that PR changed a very short list of things that we can consider safe.

**Comment by turbek** [comment]
Ah okay. Google search on the topic heavily suggested it's Gatsby Cloud related :).

Will there be a 'mark as safe' option for plugins or local plugins? It would be great if we are not losing Gatsby's functionality when we are not modifying the generated files.
